Mathlib/AlgebraicTopology/FundamentalGroupoid/Basic.lean

```diff
@@ -273,109 +248,62 @@ def FundamentalGroupoid (X : Type u) := X
 
 namespace FundamentalGroupoid
 
-instance {X : Type u} [h : Inhabited X] : Inhabited (FundamentalGroupoid X) := h
+scoped notation "πₓ" => FundamentalGroupoid
+
+/-- Help the typechecker by converting a point in a groupoid back to a point in
+the underlying topological space. -/
+def toTop {X} (x : πₓ X) : X := x
+#align fundamental_groupoid.to_top FundamentalGroupoid.toTop
 
-attribute [reducible] FundamentalGroupoid
+/-- Help the typechecker by converting a point in a topological space to a
+point in the fundamental groupoid of that space. -/
+def fromTop {X} (x : X) : πₓ X := x
+#align fundamental_groupoid.from_top FundamentalGroupoid.fromTop
+
+instance {X : Type u} [h : Inhabited X] : Inhabited (FundamentalGroupoid X) := h
 
 attribute [local instance] Path.Homotopic.setoid
 
 instance : CategoryTheory.Groupoid (FundamentalGroupoid X) where
-  Hom x y := Path.Homotopic.Quotient x y
-  id x := ⟦Path.refl x⟧
-  comp {x y z} := Path.Homotopic.Quotient.comp
-  id_comp {x y} f :=
-    Quotient.inductionOn f fun a =>
-      show ⟦(Path.refl x).trans a⟧ = ⟦a⟧ from Quotient.sound ⟨Path.Homotopy.reflTrans a⟩
-  comp_id {x y} f :=
-    Quotient.inductionOn f fun a =>
-      show ⟦a.trans (Path.refl y)⟧ = ⟦a⟧ from Quotient.sound ⟨Path.Homotopy.transRefl a⟩
-  assoc {w x y z} f g h :=
-    Quotient.inductionOn₃ f g h fun p q r =>
-      show ⟦(p.trans q).trans r⟧ = ⟦p.trans (q.trans r)⟧ from
-        Quotient.sound ⟨Path.Homotopy.transAssoc p q r⟩
-  inv {x y} p :=
-    Quotient.lift (fun l : Path x y => ⟦l.symm⟧)
-      (by
-        rintro a b ⟨h⟩
-        simp only
-        rw [Quotient.eq]
-        exact ⟨h.symm₂⟩)
-      p
-  inv_comp {x y} f :=
-    Quotient.inductionOn f fun a =>
-      show ⟦a.symm.trans a⟧ = ⟦Path.refl y⟧ from
-        Quotient.sound ⟨(Path.Homotopy.reflSymmTrans a).symm⟩
-  comp_inv {x y} f :=
-    Quotient.inductionOn f fun a =>
-      show ⟦a.trans a.symm⟧ = ⟦Path.refl x⟧ from
-        Quotient.sound ⟨(Path.Homotopy.reflTransSymm a).symm⟩
+  Hom x y := Path.Homotopic.Quotient (toTop x) (toTop y)
+  id x := ⟦Path.refl (toTop x)⟧
```

Comment:
After making `FundamentalGroupoid` reducible, `toTop`s are inserted to make it typecheck.